cibyl/cli/parser.py

```diff
@@ -99,6 +101,21 @@ def print_help(self) -> None:
         arguments that are currently added."""
         self.argument_parser.print_help()
 
+    def add_argument_to_tree(self, arg: Argument,
+                             parent_queries: Set[str]) -> None:
+        """Add argument to the tree of query method relationships. This tree
+        relates query method that have dependencies, e.g. get_jobs and
+        get_builds. Queries are added as nodes and the edges of the graph marks
+        the relationships. Nodes cannot be duplicated and self-links are
+        avoided."""
+        if arg.func in parent_queries:
+            # avoid creating self-links
+            return
```

Comment:
If I am 'tests' and in 'parent_queries' I give you: 'jobs', 'builds' and 'tests'. I think that this condition will lose 'jobs' and 'builds'. Maybe you could add a continue condition on the for loop or raise an error here because it should not happen.

I also do not know how likely that is to happen.

Reply:
I think it's quite unlikely, but I can rule it out. As you say, moving this check into the loop would solve the problem in the future if it would ever arise, so I'll do it, thank for pointing it out.